velox/functions/lib/window/NthValueBase.h

```diff
@@ -96,6 +64,23 @@ class NthValueFunction : public exec::WindowFunction {
     partitionOffset_ += numRows;
   }
 
+ protected:
+  void setOffsetIndex(const column_index_t offsetIndex) {
+    offsetIndex_ = offsetIndex;
+  }
+
+  void setConstantOffset(const std::optional<vector_size_t>& constantOffset) {
```

Comment:
Do we need to have 2 separate functions for setConstantOffset and setIsConstantOffsetNull ? Since constantOffset is anyways optional we use it to set isConstantOffsetNull_ if the optional is not there.

Reply:
Updated, thanks.